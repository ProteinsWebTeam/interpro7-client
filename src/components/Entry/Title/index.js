import React, {PropTypes as T} from 'react';

import TypeTag from 'components/TypeTag';
import {
  Name, OriginDB, SourceOrganism,
} from 'components/SimpleCommonComponents';

import styles from 'styles/blocks.css';

const Title = ({metadata, pathname}) => (
  <div className={styles.card}>
    <h2>{metadata.name.name}</h2>
    <div>
      {metadata.type && <TypeTag type={metadata.type} full={true} />}
      <Name name={metadata.name} accession={metadata.accession} />
      <OriginDB source={metadata.source_database} pathname={pathname} />
      {metadata.gene && <p>Gene: {metadata.gene}</p>}
      {
        metadata.source_organism &&
        <SourceOrganism {...metadata.source_organism} />
      }
    </div>
  </div>
);
Title.propTypes = {
  metadata: T.shape({
    name: T.object.isRequired,
    type: T.string,
    accession: T.string.isRequired,
    source_database: T.string.isRequired,
    gene: T.string,
    source_organism: T.object,
  }).isRequired,
  pathname: T.string.isRequired,
};

export default Title;
