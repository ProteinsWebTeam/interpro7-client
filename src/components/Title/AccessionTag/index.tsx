import React from 'react';
import config from 'config';

import cssBinder from 'styles/cssBinder';
import styles from '../style.css';

const css = cssBinder(styles);

const accessionDisplay = new Set(['protein', 'structure', 'proteome', 'set']);

const mapNameToClass = new Map([
  ['domain', 'title-id-domain'],
  ['family', 'title-id-family'],
  ['repeat', 'title-id-repeat'],
  ['unknown', 'title-id-unknown'],
  ['conserved_site', 'title-id-site'],
  ['binding_site', 'title-id-site'],
  ['active_site', 'title-id-site'],
  ['ptm', 'title-id-site'],
  ['homologous_superfamily', 'title-id-hh'],
]);

type Props = {
  accession?: string;
  db?: string;
  type?: string;
  mainType: string;
  isIPScanResult: boolean;
};

const lightFontDBs = ['ssf', 'pfam'];

const AccessionTag = ({
  accession,
  db,
  type,
  mainType,
  isIPScanResult = false,
}: Props) => {
  const isEntry = mainType === 'entry';
  return (
    <div className={css('title-id')} data-testid="accession">
      {
        // Red, Green for domains,  Purple for sites, and Blue for Homologous accession: for InterPro page only
        isEntry && type && db && db.toLowerCase() === 'interpro' && (
          <span className={css(mapNameToClass.get(type))}>{accession}</span>
        )
      }
      {
        // Blue accession: for Member Database and Unknown entry-type
        isEntry && type && db && db.toLowerCase() !== 'interpro' && (
          <span
            style={{
              backgroundColor: config.colors.get(db),
            }}
            className={css('title-id-md', { white: lightFontDBs.includes(db) })}
          >
            {accession}
          </span>
        )
      }
      {
        // greyblueish accession: for protein , structure, and proteomes and no accession for tax
        accessionDisplay.has(mainType) &&
          db !== 'taxonomy' &&
          !isIPScanResult && (
            // for proteins, structures and proteomes (no accession in title for taxonomy and sets)
            <span className={css('title-id-other')}>{accession}</span>
          )
      }
    </div>
  );
};

export default AccessionTag;
