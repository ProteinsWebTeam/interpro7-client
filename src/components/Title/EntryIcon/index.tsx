import React from 'react';
import Tooltip from 'components/SimpleCommonComponents/Tooltip';

import cssBinder from 'styles/cssBinder';
import ipro from 'styles/icons.css';

const css = cssBinder(ipro);

const EntryIcon = ({ type }: { type: string }) => (
  <Tooltip title={`Type: ${type.replace('_', ' ')}`}>
    <interpro-type
      type={type.replace('_', ' ')}
      dimension="4em"
      aria-label="Entry type"
      data-testid="entry-type-icon"
    >
      {
        // IE11 fallback for icons
      }
      <span
        className={css('icon-type', {
          ['icon-family']: type === 'family',
          ['icon-domain']: type === 'domain',
          ['icon-repeat']: type === 'repeat',
          ['icon-hh']: type.replace('_', ' ') === 'homologous superfamily',
          ['icon-site']:
            type.replace('_', ' ') === 'conserved site' ||
            type.replace('_', ' ') === 'binding site' ||
            type.replace('_', ' ') === 'active site' ||
            type === 'ptm',
        })}
      >
        {type === 'family' ? 'F' : null}
        {type === 'domain' ? 'D' : null}
        {type === 'repeat' ? 'R' : null}
        {type.replace('_', ' ') === 'homologous superfamily' ? 'H' : null}
        {type.replace('_', ' ') === 'conserved site' ||
        type.replace('_', ' ') === 'binding site' ||
        type.replace('_', ' ') === 'active site' ||
        type === 'ptm'
          ? 'S'
          : null}
      </span>
    </interpro-type>
  </Tooltip>
);

export default EntryIcon;
