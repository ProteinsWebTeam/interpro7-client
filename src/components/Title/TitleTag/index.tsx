import React from 'react';
import Tooltip from 'components/SimpleCommonComponents/Tooltip';
import TooltipAndRTDLink from 'components/Help/TooltipAndRTDLink';

import cssBinder from 'styles/cssBinder';
import styles from '../style.css';
import fonts from 'EBI-Icon-fonts/fonts.css';

const css = cssBinder(styles, fonts);

const rtdLinks = {
  entry: {
    interpro: 'entry-page',
    dbs: 'member-database-page',
  },
  protein: 'protein-entry-page',
  structure: 'structure-entry-page',
  taxonomy: 'taxonomy-entry-page',
  proteome: 'proteome-entry-page',
  set: 'set-entry-page',
};
type Props = {
  db: string;
  mainType: Endpoint;
  dbLabel: string;
};

const TitleTag = ({ db, mainType, dbLabel }: Props) => {
  const isEntry = mainType === 'entry';
  const isInterPro = db && db.toLowerCase() === 'interpro';
  let rtdLink = '';
  if (isEntry) {
    rtdLink = rtdLinks.entry[isInterPro ? 'interpro' : 'dbs'];
  } else {
    rtdLink = (rtdLinks?.[mainType.toLowerCase() as Endpoint] as string) || '';
  }
  return (
    <div className={css('title-tag')} data-testid="title">
      {db && (
        <div
          className={css('tag', {
            secondary: !isEntry || isInterPro,
            'md-p': isEntry && !isInterPro,
          })}
        >
          {dbLabel} {mainType}
          {
            // Set
            mainType === 'set' && (
              <Tooltip title="A Set is defined as a group of related entries">
                {' '}
                <span
                  className={css('small', 'icon', 'icon-common')}
                  data-icon="&#xf129;"
                />
              </Tooltip>
            )
          }
        </div>
      )}{' '}
      <TooltipAndRTDLink rtdPage={`browse.html#${rtdLink}`} />
    </div>
  );
};

export const FragmentTag = ({ isFragment }: { isFragment: boolean }) => {
  if (!isFragment) return null;
  return (
    <div className={css('fragment-tag')} data-testid="title">
      <div className={css('tag')}>Fragment</div>
    </div>
  );
};

export default TitleTag;
