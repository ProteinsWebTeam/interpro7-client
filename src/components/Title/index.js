import React, { PureComponent } from 'react';
import T from 'prop-types';
import { Helmet } from 'react-helmet';
import MemberSymbol from 'components/Entry/MemberSymbol';
import Tooltip from 'components/SimpleCommonComponents/Tooltip';
import loadWebComponent from 'utils/load-web-component';

import { foundationPartial } from 'styles/foundation';

import fonts from 'EBI-Icon-fonts/fonts.css';
import ipro from 'styles/interpro-new.css';
import styles from './style.css';
import loadData from '../../higherOrder/loadData';
import { getUrlForMeta } from '../../higherOrder/loadData/defaults';

import config from 'config';

const f = foundationPartial(fonts, ipro, styles);

const mapNameToClass = new Map([
  ['Domain', 'title-id-domain'],
  ['Family', 'title-id-family'],
  ['Repeat', 'title-id-repeat'],
  ['Unknown', 'title-id-unknown'],
  ['Conserved_site', 'title-id-site'],
  ['Binding_site', 'title-id-site'],
  ['Active_site', 'title-id-site'],
  ['PTM', 'title-id-site'],
  ['Homologous_superfamily', 'title-id-hh'],
]);

/*:: type Props = {
  metadata: {
    name: { name: string, short: ?string },
    accession: string | number,
    source_database?: string,
    type?: string,
    gene?: string,
    experiment_type?: string,
    source_organism?: Object,
    release_date?: string,
    chains?: Array<string>,
  },
  mainType: string,
  data?: Object,
}; */

const accessionDisplay = new Set(['protein', 'structure', 'organism']);

class Title extends PureComponent /*:: <Props> */ {
  static propTypes = {
    metadata: T.object.isRequired,
    data: T.object.isRequired,
    mainType: T.string.isRequired,
  };

  componentDidMount() {
    loadWebComponent(() =>
      import(/* webpackChunkName: "interpro-components" */ 'interpro-components').then(
        m => m.InterproType,
      ),
    ).as('interpro-type');
  }

  render() {
    const { metadata, mainType, data } = this.props;
    const isEntry = mainType === 'entry';
    const databases = data && data.payload && data.payload.databases;
    const dbLabel =
      databases && databases[metadata.source_database]
        ? databases[metadata.source_database].name
        : metadata.source_database;

    return (
      <div className={f('title')}>
        {// Entry icon
        isEntry &&
          metadata.type &&
          metadata.source_database &&
          metadata.source_database.toLowerCase() === 'interpro' && (
            <Tooltip title={`${metadata.type.replace('_', ' ')} type`}>
              <interpro-type
                type={metadata.type.replace('_', ' ')}
                dimension="4em"
              />
            </Tooltip>
          )}

        {// MD icon
        isEntry &&
          metadata.type &&
          metadata.source_database &&
          metadata.source_database.toLowerCase() !== 'interpro' && (
            <div className={f('icon-container')}>
              <MemberSymbol type={metadata.source_database} />
            </div>
          )}

        <Helmet>
          <title>{metadata.accession}</title>
        </Helmet>
        <h3>
          {metadata.name.name}{' '}
          {// Red, Green for domains,  Purple for sites, and Blue for Homologous accession: for InterPro page only
          isEntry &&
            metadata.type &&
            metadata.source_database &&
            metadata.source_database.toLowerCase() === 'interpro' && (
              <small className={f(mapNameToClass.get(metadata.type))}>
                {metadata.accession}
              </small>
            )}
          {// Blue accession: for Member Database and Unknown entry-type
          isEntry &&
            metadata.type &&
            metadata.source_database &&
            metadata.source_database.toLowerCase() !== 'interpro' && (
              <small
                style={{
                  backgroundColor: config.colors.get(metadata.source_database),
                }}
                className={f('title-id-md')}
              >
                {metadata.accession}
              </small>
            )}
          {// greyish accession: for protein , structure, and proteomes and no accession for tax
          accessionDisplay.has(mainType) &&
            metadata.source_database !== 'taxonomy' && (
              // no accession for Taxonomy but in blue for protein (reviewed), structure (pdb), and proteomes (proteome)
              <small className={f('title-id-other')}>
                {metadata.accession}
              </small>
            )}
        </h3>

        {// InterPro Entry
        isEntry &&
          metadata.type &&
          metadata.source_database &&
          metadata.source_database.toLowerCase() === 'interpro' && (
            <div className={f('tag', 'secondary')}>{dbLabel} entry</div>
          )}

        {// MD Entry -signature
        isEntry &&
          metadata.type &&
          metadata.source_database &&
          metadata.source_database.toLowerCase() !== 'interpro' && (
            <div className={f('tag', 'md-p')}>{dbLabel} Entry</div>
          )}

        {// protein page
        mainType === 'protein' && (
          <div className={f('tag', 'secondary', 'margin-bottom-large')}>
            Protein {dbLabel}
          </div>
        )}

        {// Structure
        mainType === 'structure' && (
          <div className={f('tag', 'secondary', 'margin-bottom-large')}>
            Structure
          </div>
        )}

        {// Species
        metadata.source_database !== 'proteome' &&
          mainType === 'organism' && (
            <div className={f('tag', 'secondary', 'margin-bottom-large')}>
              {dbLabel}
            </div>
          )}

        {// Proteome
        metadata.is_reference ? (
          <div className={f('tag', 'secondary', 'margin-bottom-large')}>
            Reference proteome{' '}
            <Tooltip title="Some proteomes have been (manually and algorithmically) selected as reference proteomes. They cover well-studied model organisms and other organisms of interest for biomedical research and phylogeny.">
              <span
                className={f('small', 'icon', 'icon-common')}
                data-icon="ℹ"
                aria-label="Some proteomes have been (manually and algorithmically) selected as reference proteomes. They cover well-studied model organisms and other organisms of interest for biomedical research and phylogeny."
              />
            </Tooltip>
          </div>
        ) : null}

        {// Set
        mainType === 'set' && (
          <div className={f('tag', 'secondary', 'margin-bottom-large')}>
            Set {dbLabel}{' '}
            <Tooltip title="A Set is defined as a group of related entries">
              <span
                className={f('small', 'icon', 'icon-common')}
                data-icon="ℹ"
              />
            </Tooltip>
          </div>
        )}
      </div>
    );
  }
}

export default loadData(getUrlForMeta)(Title);
