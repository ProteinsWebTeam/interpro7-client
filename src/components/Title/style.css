@import '../../styles/colors.css';

.title {
  display: flex;

  & > div:first-of-type:not(.icon-container) {
    margin-top: 0.5rem; /* when using tooltip only way to move the SVG icon*/
  }

  & interpro-type {
    margin-right: 1.5rem;
  }
  & .icon-container {
    /*negative values as the viewbox is not well defined*/
    margin: -0.7rem 0.5rem 0 -1rem;
    width: 7.6%;
    min-width: 80px;
  }
}

/* temp to test - version with notag*/
.title.notag {
  & interpro-type {
    margin: 0 1.5rem 0 0;
  }
  & .title-tag {
    display: none;
  }
  & .title-id {
    margin-left: 1.5rem;
  }
  & h3 {
    line-height: inherit;
  }
}

.title-id {
  text-transform: uppercase; /*temp ?*/
  color: #444;
  margin: 0.5rem 0 0 1.5rem;
  font-size: 130%;
}
.title-id span {
  padding: 0.2rem 0.5rem;
}

.title-id-family {
  background-color: var(--colors-family-sub);
}
.title-id-domain {
  background-color: var(--colors-domain-sub);
}
.title-id-repeat {
  background-color: var(--colors-repeat-sub);
}
.title-id-site {
  background-color: var(--colors-site-sub);
}
.title-id-hh {
  background-color: var(--colors-homolog-sub);
}
.title-id-unknown {
  background-color: var(--colors-unknown-sub);
}
/*used for protein, structure, organism page*/
.title-id-other {
  background-color: var(--colors-other-main);
}

.title-id-md {
  background-color: var(--colors-md-main);
}

.tag {
  margin-bottom: 1rem;
}

@media screen and (max-width: 75rem) {
  /*reduce SVG icon*/
  .title interpro-type {
    font-size: 86%;
    margin-right: 1rem;
  }
  .title.notag interpro-type {
    margin-right: 1rem;
  }
  /*reduce tag font*/
  .title-tag .tag {
    font-size: 70%;
  }
  .title-id {
    margin-left: 0.5rem;
    font-size: 81%;
  }
}
