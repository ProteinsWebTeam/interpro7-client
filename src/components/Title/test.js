/* eslint-env node */
/* eslint-env mocha */
import 'babel-polyfill';

import React from 'react';
import {createRenderer} from 'react-addons-test-utils';
import chai, {expect} from 'chai';
import jsxChai from 'jsx-chai';

import Link from 'components/Link';

import Title, {InterproSymbol} from '.';

import ipro from 'styles/interpro-new.css';

chai.use(jsxChai);
const renderer = createRenderer();

describe.skip('<Title />', () => {
  describe('For an entry', () => {
    it('should render a title component correctly', () => {
      renderer.render(
        <Title
          pathname="/entry/pathname/"
          metadata={{
            name: {
              name: 'Piwi domain',
            },
            type: 'domain',
            accession: 'PF02171',
            source_database: 'pfam',
          }}
        />
      );
      expect(renderer.getRenderOutput()).to.deep.equal(
        <div>
          <div className="my-svg-container">
            <InterproSymbol type="domain"/>
          </div>
          <h3>Piwi domain <small>(PF02171)</small></h3>
          <div className={ipro['md-hlight']}>
            <h5>Member database:&nbsp;
              <Link to="/entry/pfam/">
                pfam
              </Link>
            </h5>
          </div>
        </div>
      );
    });
  });
  describe('For a protein', () => {
    it('should render a title component correctly', () => {
      renderer.render(
        <Title
          pathname="/some/pathname/"
          metadata={{
            length: 462,
            description:  ["Created on Monday 13 August 2007 by user 'UJJWAL'"],
            name: {
              other: [],
              name: 'VATB_METVS',
              short: '',
            },
            gene: '',
            id: 'VATB_METVS',
            accession: 'A6UP55',
          }}
        />
      );
      expect(renderer.getRenderOutput()).to.deep.equal(
        <div>
          <h3>
            VATB_METVS
            <small>(A6UP55)</small>
          </h3>
        </div>
      );
    });
  });
  describe('For a structure', () => {
    it('should render a title component correctly', () => {
      renderer.render(
        <Title pathname="/some/pathname/1JM7/" metadata={{
          name: {
            name: 'RING-domain heterodimer',
          },
          accession: '1JM7',
          source_database: 'pdb',
          experiment_type: 'Solution NMR',
          release_date: '2009-02-24',
          chains: ['A', 'B'],
        }}
        />
      );
      expect(renderer.getRenderOutput()).to.deep.equal(
        <div>
          <h3>RING-domain heterodimer
            <small>(1JM7)</small>
          </h3>
        </div>
      );
    });
  });
});
