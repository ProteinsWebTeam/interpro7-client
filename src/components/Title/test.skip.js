import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Title from '.';

const renderer = new ShallowRenderer();

describe.skip('<Title />', () => {
  describe('For an entry', () => {
    test('should render a title component correctly', () => {
      renderer.render(
        <Title
          metadata={{
            name: {
              name: 'Piwi domain',
            },
            type: 'domain',
            accession: 'PF02171',
            source_database: 'pfam',
          }}
        />,
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
  describe('For a protein', () => {
    test('should render a title component correctly', () => {
      renderer.render(
        <Title
          metadata={{
            length: 462,
            description: ["Created on Monday 13 August 2007 by user 'UJJWAL'"],
            name: {
              other: [],
              name: 'VATB_METVS',
              short: '',
            },
            gene: '',
            id: 'VATB_METVS',
            accession: 'A6UP55',
          }}
        />,
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
  describe('For a structure', () => {
    test('should render a title component correctly', () => {
      renderer.render(
        <Title
          metadata={{
            name: {
              name: 'RING-domain heterodimer',
            },
            accession: '1JM7',
            source_database: 'pdb',
            experiment_type: 'Solution NMR',
            release_date: '2009-02-24',
            chains: ['A', 'B'],
          }}
        />,
      );
      expect(renderer.getRenderOutput()).toMatchSnapshot();
    });
  });
});
