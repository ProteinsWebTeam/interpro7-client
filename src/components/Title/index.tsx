import React, { PureComponent } from 'react';
import { Helmet } from 'react-helmet-async';

import MemberSymbol from 'components/Entry/MemberSymbol';
import loadWebComponent from 'utils/load-web-component';

import loadData from 'higherOrder/loadData/ts';
import { getUrlForMeta } from '../../higherOrder/loadData/defaults';
import { createSelector } from 'reselect';
import { connect } from 'react-redux';
import { markFavourite, unmarkFavourite } from 'actions/creators';

import EntryIcon from './EntryIcon';
import AccessionTag from './AccessionTag';
import TitleTag, { FragmentTag } from './TitleTag';

import cssBinder from 'styles/cssBinder';
import fonts from 'EBI-Icon-fonts/fonts.css';
import ipro from 'styles/interpro-vf.css';
import styles from './style.css';

const css = cssBinder(fonts, ipro, styles);

type Props = {
  metadata: Metadata;
  mainType: Endpoint;
  entries?: Array<string>;
  markFavourite: typeof markFavourite;
  unmarkFavourite: typeof unmarkFavourite;
};

interface LoadedProps extends Props, LoadDataProps<RootAPIPayload> {}

export class Title extends PureComponent<LoadedProps> {
  componentDidMount() {
    loadWebComponent(() =>
      import(
        /* webpackChunkName: "interpro-components" */ 'interpro-components'
      ).then((m) => m.InterproType)
    ).as('interpro-type');
  }

  manageFavourites(metadata: Metadata) {
    if ((this.props.entries || []).includes(metadata.accession)) {
      this.props.unmarkFavourite(metadata.accession);
    } else {
      this.props.markFavourite(metadata.accession, { metadata });
    }
  }

  // eslint-disable-next-line complexity
  render() {
    const { metadata, mainType, data } = this.props;
    const isEntry = mainType === 'entry';
    const databases = data && data.payload && data.payload.databases;
    const dbLabel =
      databases && databases[metadata.source_database]
        ? databases[metadata.source_database].name
        : metadata.source_database;

    const longName =
      typeof (metadata as ProteinMetadata).name === 'string'
        ? (metadata as ProteinMetadata).name
        : (metadata as EntryMetadata).name.name ||
          (metadata as EntryMetadata).name.short ||
          (metadata as EntryMetadata).accession;
    const isIPScanResult = longName === 'InterProScan Search Result';

    const entryMetadata = metadata as EntryMetadata;

    return (
      <div className={css('title')} data-testid="titlebar">
        {isEntry &&
          entryMetadata.type &&
          entryMetadata &&
          entryMetadata.source_database.toLowerCase() === 'interpro' && (
            <>
              <EntryIcon type={(metadata as EntryMetadata).type} />
              <Helmet
                titleTemplate={`${longName} ${metadata.accession} - Entry - InterPro`}
              >
                <title>InterPro</title>
              </Helmet>
            </>
          )}

        {
          // MD icon
          isEntry &&
            entryMetadata.type &&
            entryMetadata.source_database &&
            entryMetadata.source_database.toLowerCase() !== 'interpro' && (
              <div className={css('icon-container')}>
                <MemberSymbol
                  type={entryMetadata.source_database as MemberDB}
                  svg={false}
                  filter={false}
                  includeLink={true}
                />
              </div>
            )
        }
        {metadata && (
          <Helmet
            titleTemplate={`${longName} (${metadata.accession}) - ${
              isEntry ? dbLabel : ' '
            } ${mainType} - InterPro`}
          >
            <title>InterPro</title>
          </Helmet>
        )}

        <div
          className={css('title-name', {
            ipscan: isIPScanResult,
          })}
        >
          <div className={css('title-fav')}>
            <AccessionTag
              accession={metadata.accession}
              db={metadata.source_database}
              type={(metadata as EntryMetadata).type}
              mainType={mainType}
              isIPScanResult={isIPScanResult}
            />

            <h3
              className={css({
                'margin-bottom-large': isIPScanResult,
              })}
            >
              {longName}{' '}
            </h3>
            {
              // Showing Favourites only for InterPro entries for now
              isEntry &&
                entryMetadata.type &&
                entryMetadata.source_database &&
                entryMetadata.source_database.toLowerCase() === 'interpro' && (
                  <span
                    className={css('fav-icon')}
                    role="button"
                    onClick={() => this.manageFavourites(metadata)}
                    onKeyDown={() => this.manageFavourites(metadata)}
                    tabIndex={0}
                  >
                    <span
                      className={css(
                        'icon',
                        'icon-common',
                        (this.props.entries || []).includes(metadata.accession)
                          ? 'favourite'
                          : 'normal'
                      )}
                      data-icon="&#xf005;"
                    />
                  </span>
                )
            }
          </div>

          <TitleTag
            db={metadata?.source_database}
            mainType={mainType}
            dbLabel={dbLabel}
          />
        </div>
        <FragmentTag
          isFragment={!!(metadata as ProteinMetadata)?.is_fragment}
        />
      </div>
    );
  }
}
const mapStateToProps = createSelector(
  (state: { favourites: { entries: string[] } }) => state.favourites.entries,
  (entries: string[]) => ({ entries })
);

export default connect(mapStateToProps, { markFavourite, unmarkFavourite })(
  loadData(getUrlForMeta)(Title)
);
