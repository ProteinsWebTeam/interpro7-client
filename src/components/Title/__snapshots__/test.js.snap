// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> For a protein should render a title component correctly 1`] = `
<div
  className="title"
  data-testid="titlebar"
>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
    prioritizeSeoTags={false}
    titleTemplate="VATB_METVS (A6UP55) -   undefined - InterPro"
  >
    <title>
      InterPro
    </title>
  </Helmet>
  <div
    className="title-name"
  >
    <div
      className="title-fav"
    >
      <h3
        className=""
      >
        VATB_METVS
         
      </h3>
    </div>
    <TitleTag />
  </div>
  <AccessionTag
    accession="A6UP55"
    isIPScanResult={false}
  />
  <FragmentTag
    isFragment={false}
  />
</div>
`;

exports[`<Title /> For a structure should render a title component correctly 1`] = `
<div
  className="title"
  data-testid="titlebar"
>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
    prioritizeSeoTags={false}
    titleTemplate="RING-domain heterodimer (1JM7) -   undefined - InterPro"
  >
    <title>
      InterPro
    </title>
  </Helmet>
  <div
    className="title-name"
  >
    <div
      className="title-fav"
    >
      <h3
        className=""
      >
        RING-domain heterodimer
         
      </h3>
    </div>
    <TitleTag
      db="pdb"
      dbLabel="pdb"
    />
  </div>
  <AccessionTag
    accession="1JM7"
    db="pdb"
    isIPScanResult={false}
  />
  <FragmentTag
    isFragment={false}
  />
</div>
`;

exports[`<Title /> For an entry should render a title component correctly 1`] = `
<div
  className="title"
  data-testid="titlebar"
>
  <Helmet
    defer={true}
    encodeSpecialCharacters={true}
    prioritizeSeoTags={false}
    titleTemplate="Piwi domain (PF02171) -   undefined - InterPro"
  >
    <title>
      InterPro
    </title>
  </Helmet>
  <div
    className="title-name"
  >
    <div
      className="title-fav"
    >
      <h3
        className=""
      >
        Piwi domain
         
      </h3>
    </div>
    <TitleTag
      db="pfam"
      dbLabel="pfam"
    />
  </div>
  <AccessionTag
    accession="PF02171"
    db="pfam"
    isIPScanResult={false}
    type="domain"
  />
  <FragmentTag
    isFragment={false}
  />
</div>
`;
